htdocs/product/class/product.class.php

```diff
@@ -924,7 +924,23 @@ function delete($id=0, $notrigger=0)
                 if ($result < 0) { $error++; }
                 // End call triggers
 			}
-
+			
+			// Delete from product_batch on product delete	
+			if (! $error && $this->hasbatch())
```

Comment:
The test $this->hasbatch() should be removed.
If flag "need serial/lot number" is on, it will be true
And if flag "need serial/lot number", we must also do the delete even if there is a high probability that table is empty.
Because flag may have been modified after beeing set. 
Don't you think so ?